### PR TITLE
Module 4 - Dynamo db integration

### PR DESCRIPTION
**Pull request by ammb95**
## What was Done

- database creation and population
- integration of db with existing functions
- creation of new function createProducts

 ## Deployed Lambdas URLs

- getProductsList: https://wm2da067qj.execute-api.us-east-1.amazonaws.com/dev/products/
- getProductById: https://wm2da067qj.execute-api.us-east-1.amazonaws.com/dev/products/{id}
- createProducts: https://wm2da067qj.execute-api.us-east-1.amazonaws.com/dev/products/

### CODE REVIEW

File: products-service/src/functions/get-product-by-id/handler.ts [hunk 1]
```diff
@@ -1,25 +1,55 @@
...
+      TableName: process.env.STOCKS_TABLE,
+      KeyConditionExpression: 'ProductId = :productId',
+      ExpressionAttributeValues: {
+        ':productId': product.Id
+      }
+    };
+
+    const stockResult = await dynamodb.query(stockParams).promise();
+
+    product.count = stockResult.Items[0].Count;
```
Comment (Tanya-atatakai) [inline products-service/src/functions/get-product-by-id/handler.ts:42]: I noticed you set `product.Count` in products list (the field uses capitalisation) but here is just `product.count` (field is in lower case), so the result model here differs from products list, maybe you'd like them to be the same

File: products-service/src/functions/create-product/handler.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+const createProduct: ValidatedEventAPIGatewayProxyEvent<
+  typeof schema
+> = async event => {
+  const dynamodb = new AWS.DynamoDB.DocumentClient();
+
+  try {
+    const productId = randomUUID();
+    const params = {
+      TableName: process.env.PRODUCTS_TABLE,
+      Item: { Id: productId, ...event.body }
```
Comment (Tanya-atatakai) [inline products-service/src/functions/create-product/handler.ts:17]: hey sorry I probably broke your table a bit, I tried to create a product with postman like attached using form data and raw data, but the result is not correct, did I do something wrong? sorry again
image
image
image

Comment (ammb95) [reply products-service/src/functions/create-product/handler.ts:17]: No worries. You are supposed to send the body as json only

You have tried to send it as params

See how the url gets

image

File: products-service/src/functions/get-product-by-id/handler.ts (hunk 1 above)
Comment (ammb95) [reply products-service/src/functions/get-product-by-id/handler.ts:42]: Addressed

**Review by Tanya-atatakai — APPROVED** [review]
Thanks @ammb95 All seems to be good now, regarding postman I just forgot how to use it and didn't provide needed header :D now I see it works

it's great that you used scripts for creating and populating DBs, also what you used for random uuid helped me to understand faster how to fix 'uuid' package not working in my HW

**Pull request merged**